trl/trainer/cpo_trainer.py

```diff
@@ -936,18 +939,20 @@ def evaluation_loop(
 
             policy_output_decoded = self.generate_from_model(self.model, random_batch)
 
-            self.log(
-                {
-                    "game_log": wandb.Table(
-                        columns=["Prompt", "Policy"],
-                        rows=[
-                            [prompt, pol[len(prompt) :]]
-                            for prompt, pol in zip(random_batch["prompt"], policy_output_decoded)
-                        ],
-                    )
-                }
+            table = pd.DataFrame(
+                columns=["Prompt", "Policy"],
+                data=[
+                    [prompt, pol[len(prompt) :]] for prompt, pol in zip(random_batch["prompt"], policy_output_decoded)
+                ],
```

Comment:
for the record, this doesn't work, because the `pol` can be over-truncated. See 

<img width="1328" alt="Screenshot 2024-12-28 at 18 29 58" src="https://github.com/user-attachments/assets/81c67d07-41e0-4d42-b2ea-39939325e7e8" />

can be reproduced with 

```python
from datasets import load_dataset
from trl import CPOConfig, CPOTrainer
from transformers import AutoModelForCausalLM, AutoTokenizer

model = AutoModelForCausalLM.from_pretrained("Qwen/Qwen2-0.5B-Instruct")
tokenizer = AutoTokenizer.from_pretrained("Qwen/Qwen2-0.5B-Instruct")
train_dataset = load_dataset("trl-lib/ultrafeedback_binarized", split="train[:1%]")
eval_dataset = load_dataset("trl-lib/ultrafeedback_binarized", split="test[:1%]")

training_args = CPOConfig(output_dir="Qwen2-0.5B-CPO", logging_steps=10,  generate_during_eval=True, eval_steps=2, eval_strategy="steps")
trainer = CPOTrainer(model=model, args=training_args, processing_class=tokenizer, train_dataset=train_dataset, eval_dataset=eval_dataset)
trainer.train()
```


But this is out of the scope of this PR.